src/message/getHtmlPieceDescriptors.js

```diff
@@ -12,7 +12,9 @@ export default (
 
   let prevMessage = undefined;
   const pieces: HtmlPieceDescriptor[] = [];
-  messages.forEach(message => {
+  for (let i = 0; i < messages.length; i++) {
```

Comment:
This commit message:

```
 getHtmlPieceDescriptors [nfc]: Use a for loop. 
```

Should mention why you're doing this. Or just squash it into the next commit :)

Reply:
Hmm, yeah. Greg made it a `for` loop in his WIP commits, so I kept that, but it'd totally be possible to keep a `forEach` because the callback receives the index of the item. I'm inclined to go back to `forEach` (not `flatMap` though) in my next revision; @gnprice, what do you think?

Reply:
> Or just squash it into the next commit :)

I've opted not to do this; I think it might have muddled the next commit more than it clarified, and it would leave the change unexplained.

Reply:
The loop structure in the new revision LGTM. (I haven't looked at what the commit sequence looked like at the start of this subthread.)

```js
  messages.forEach((message, i, arr) => {
    const prevMessage = arr[i - 1];
```

It's just a little awkward because `message` and `prevMessage` aren't gotten in a symmetrical way: we get the latter from indexing into `arr` using `i`, but the former directly as an argument to the callback. It's implicit that `message` is the same as `arr[i]`, but the parallelism between that and `arr[i - 1]` doesn't appear explicitly.

I think that's why I went for a plain `for` loop in my demo. But this is fine too.

Reply:
Oh OK, got it! Thanks for that explanation; it makes sense.